src/main/java/uk/ac/ox/oxfish/fisher/equipment/fads/Fad.java

```diff
@@ -47,25 +47,31 @@
     private static final double BUOY_VALUE = 1000.0; // Buoy value in dollars, TODO: should be a parameter
 
     private final FadManager owner;
+    private final TripRecord tripDeployed;
     private final BiomassLocalBiology biology;
-    final private double[] attractionRates; // proportion of underlying biomass attracted per day
-    final private double fishReleaseProbability; // daily probability of releasing fish from the FAD
-
+    private final double[] attractionRates; // proportion of underlying biomass attracted per day
+    private final double fishReleaseProbability; // daily probability of releasing fish from the FAD
+    private final int stepDeployed;
     public Fad(
         FadManager owner,
         BiomassLocalBiology biology,
-        ImmutableMap<Species, Double> attractionRates,
-        double fishReleaseProbability
+        Map<Species, Double> attractionRates,
+        double fishReleaseProbability,
+        final int stepDeployed
     ) {
         this.owner = owner;
+        this.tripDeployed = owner.getFisher().getCurrentTrip();
         this.biology = biology;
         this.attractionRates = new double[biology.getCurrentBiomass().length];
+        this.stepDeployed = stepDeployed;
         attractionRates.forEach((species, rate) ->
             this.attractionRates[species.getIndex()] = rate
         );
         this.fishReleaseProbability = fishReleaseProbability;
     }
 
+    public TripRecord getTripDeployed() { return tripDeployed; }
```

Comment:
is there a reason why this is held in the FAD rather than the trip record itself?

Reply:
Do you mean adding a list of FADs deployed to the trip record? It just hadn't occurred to me. I usually try to avoid adding FAD specific stuff to POSEIDON's core classes.

Reply:
Your call. No problem either way for me.